src/discovery/model.ts

```diff
@@ -100,7 +102,7 @@ export interface IDiscoveryNetwork {
     close(): void;
 
     /** Request devices on the network to identify themselves */
-    ping(): Promise<void>;
+    ping(deviceIp: string | undefined): Promise<void>;
```

Comment:
As above:

```suggestion
    ping(deviceIp?: string): Promise<void>;
```

Also, since the behavior is different if a `deviceIp` is provided, we should probably indicate that in the documentation, something like 

```
    /**
      * Request devices on the network to identify themselves. A specific `deviceIp`
      * may be provided to instead talk to a specific device
      */
```